packages/calypso-e2e/src/lib/utils/get-test-account-by-feature.ts

```diff
@@ -0,0 +1,124 @@
+type Env = 'edge' | 'stable';
+
+type SiteType = 'simple' | 'atomic';
+
+type Variant = 'siteEditor';
+
+type Feature = 'gutenberg' | 'coblocks';
+type FeatureKey = { [ key in Feature ]?: Env | undefined } & {
+	siteType: SiteType;
+	variant?: Variant;
+};
+export type FeatureCriteria = FeatureKey & { accountName: string };
+type FeatureMap = Map< string, string >;
+
+/**
+ * Sort the keys of the `FeatureKey` structure and finally converts it
+ * into a string representation that can be used as a key to a `Map`.
+ *
+ * @param {FeatureKey} o the FeatureKey object
+ * @returns {string} a string representation of the key
+ */
+function stringifyKey( o: FeatureKey ) {
+	const keys = Object.keys( o ) as [ keyof FeatureKey ];
+	const sorted = keys.sort().reduce( ( sorted, key ) => {
+		sorted[ key ] = o[ key ];
+		return sorted;
+	}, {} as any ) as FeatureKey;
+
+	return JSON.stringify( sorted );
+}
+
+/**
+ * Turns an array of `FeastureCriteria` into a Map that can be easily queried
+ * with stringified `FeatureKey`s.
+ *
+ * @param {FeatureCriteria[]} criteria An array of `FeatureCriteria` objects.
+ * @param {FeatureMap} map Instance of `FeatureMap` that will hold the table  queriable by
+ * (stringified)`FeatureKey`s objects.
+ * @returns {FeatureMap} A `Map`object built from the `criteria` that can be queried by a
+ * (stringified) `FeatureKey` object.
+ */
+function criteriaToMap( criteria: FeatureCriteria[], map: FeatureMap ): FeatureMap {
+	return criteria.reduce( ( acc, criteria ) => {
```

Comment:
Tiny optional style nit I should have mentioned before, but stylistically I always prefer to give these args in callbacks more clear names, even though they're part of commonly known signatures. Can we call this something like `currentMap` or something similar?

Reply:
Sure! 👍🏻